nixos/modules/services/networking/tailscale.nix

```diff
@@ -31,6 +31,12 @@ in {
 
     package = lib.mkPackageOptionMD pkgs "tailscale" {};
 
+    openFirewall = mkOption {
+      default = false;
```

Comment:
I'm not sure about nixos convention, but: could we make this true by default? The port in question only serves WireGuard and another cryptographic UDP protocol with similar semantics, which means unless you have the correct keys to talk to it, it will not respond to you in any way.

So, it's unusual in the sense that opening the port in the firewall has only benefits: it improves Tailscale connectivity for this machine, but does not expose anything unexpectedly to third parties.

Reply:
I did explicitly stick to the convention of not opening the firewall by default per that guideline. That said, it was an initial surprise as a user to enable Tailscale, and everything was talking through relays. I figured having the option like many other services would serve as a reminder, without needing to change the default.

Also, I wasn't going to break a key NixOS convention in my first contribution. 😄

Reply:
I think the convention is fine in general, but the _intent_ of that convention doesn't make sense in Tailscale's case because there is no downside to opening the firewall, whereas most other services, firewalling is an important admin decision.

That said... nixos users are already used to telling every service they run to open the firewall, so it's not like it's going to be surprising for people :shrug: Someday I would like to have the fight to open the firewall by default for tailscale, but having the option as a first step and sticking to the general convention sounds fine.

Congrats on your first NixOS PR by the way! Welcome aboard, and thanks for fixing stuff for everyone :)